Source/Wikitivity.Agent/WikitivityUploadsAgent.cs

```diff
@@ -51,155 +57,130 @@ public override void Execute()
+							// Create a list holding the entirety of the operations for the agent to perform.
+							List<SingleRequestObject> completeListByRequestID = new List<SingleRequestObject>();
+							foreach (var singleArticleResult in getWorkloadResultSlim.Objects)
+							{
+								completeListByRequestID.Add(new SingleRequestObject
 								{
-									ArtifactID = InitJobArtiID,
-									PageTitle = InitPageTitle,
-									RequestID = InitRequestID,
-									RequestUrl = InitRequestUrl
-								};
 
-								//Now I want to query for all requests with the same ID
 
-								Query<RDO> getArticlesByRequestID = new Query<RDO>();
-								getArticlesByRequestID.Condition = new TextCondition("Request ID",
-									TextConditionEnum.EqualTo,
-									seedRequest.RequestID);
-								getArticlesByRequestID.Fields = FieldValue.AllFields;
-								getArticlesByRequestID.ArtifactTypeGuid = WikitivityRDOGUID;
+									ArtifactID = singleArticleResult.ArtifactID,
+									PageTitle = singleArticleResult.Values[2].ToString(),
+									RequestID = singleArticleResult.Values[1].ToString(),
+									RequestUrl = singleArticleResult.Values[0].ToString(),
+									ControlNumVal = singleArticleResult.Values[3].ToString(),
 
-								var ArticlesByRequestIDResultSet =
-									proxy.Repositories.RDO.Query(
-										getArticlesByRequestID); //this resultset will hold the entirety of the requested operations
-
-								RaiseMessage("Retrieved full list of jobs" + ArticlesByRequestIDResultSet.TotalCount, 1);
-
-
-								// Create a list holding the entirety of the operations for the agent to perform.
-								List<SingleRequestObject> completeListByRequestID = new List<SingleRequestObject>();
-								foreach (var singleArticleResult in ArticlesByRequestIDResultSet.Results)
+								});
+							}
+							RaiseMessage(
+								"Creation of master list complete - Total Count of Articles " +
+								completeListByRequestID.Count, 1);
+							//I now have a full list of articles for a specific request ID stored in CompleteListByRequestID
+							//The next step is to batch them for more manageable agent functionality
+						
+							List<SingleRequestObject> batchedRequestList = new List<SingleRequestObject>();
+							while (completeListByRequestID.Count >= 1)
+							{
+								try
 								{
-									var singleArticle = singleArticleResult.Artifact;
-									completeListByRequestID.Add(new SingleRequestObject
-									{
-										ArtifactID = singleArticle.ArtifactID,
-										PageTitle = singleArticle.Fields.Get("Page Title").ValueAsLongText,
-										RequestID = singleArticle.Fields.Get("Request ID").ValueAsLongText,
-										RequestUrl = singleArticle.Fields.Get("Request Url").ValueAsLongText,
-										ControlNumVal = singleArticle.Fields.Get("Name").ValueAsLongText
+									//Steps:
+									// 1. I need to create my batch set (batchedRequestList)
+									batchedRequestList = completeListByRequestID.Take(50).ToList();

```

Comment:
My gut is if you have comments that are "Steps"  this indicates a need for a new class, with a method to call all the functions and each step turned into a function.